### PR TITLE
feat(admin): add a service search filter on facility resources list

### PR DESCRIPTION
**Pull request by xflord**
* user can now filter resources by services that are assigned to those resources

requires that https://github.com/CESNET/perun/pull/3632 is merged into backend

### CODE REVIEW

**Review by xkostka2 — CHANGES_REQUESTED** [review]
Filtering by service does not work.

File: apps/admin-gui/src/app/facilities/pages/facility-detail-page/facility-resources/facility-resources.component.ts [hunk 1]
```diff
@@ -109,4 +135,10 @@ export class FacilityResourcesComponent implements OnInit {
       }
     });
   }
+
+  serviceSelected(service: Service): void {
+    this.selectedService = service;
+    this.refreshTable();
+    this.cd.detectChanges();
```
Comment (bodnara) [inline apps/admin-gui/src/app/facilities/pages/facility-detail-page/facility-resources/facility-resources.component.ts:142]: Don't think we need to detect changes here.

**Review by bodnara — CHANGES_REQUESTED** [review]
I think the service select field is a little bit too wide while the filter filed is quite small. Both could be adjusted. Also the margin between select and filter is inconsitent with other elements in the row.
image

The `serviceSlected` method is triggered twice each time service is selected which results in the table refreshing twice as well.

File: apps/admin-gui/src/app/facilities/pages/facility-detail-page/facility-resources/facility-resources.component.ts (hunk 1 above)
Comment (xflord) [reply apps/admin-gui/src/app/facilities/pages/facility-detail-page/facility-resources/facility-resources.component.ts:142]: Rarely a 'Expression has changed after it was checked' error was thrown when refresing the table. I couldn't quite figure out why but this fixed it

**Review by bodnara — CHANGES_REQUESTED** [review]
So the solution for the double trigger is valid and we use this on multiple places across the app. But maybe we could solve the root of this problem so no one else has to suffer. The culprit in this case is inside of `entity-search-select.component.ts`.  Specifically the `entitiesCtrl`, we are subscribing to value changes and immediately emitting new value. A better option is to emit only when the value is different than the previous one using the `distinctUntilChanged` operator.

**Comment by xflord** [comment]
> So the solution for the double trigger is valid and we use this on multiple places across the app. But maybe we could solve the root of this problem so no one else has to suffer. The culprit in this case is inside of `entity-search-select.component.ts`. Specifically the `entitiesCtrl`, we are subscribing to value changes and immediately emitting new value. A better option is to emit only when the value is different than the previous one using the `distinctUntilChanged` operator.

Will do

**Review by bodnara — APPROVED** [review]

**Review by xkostka2 — APPROVED** [review]

**Pull request merged**